### PR TITLE
stratis-min

### PR DESCRIPTION
**Pull request by jbaublitz**
Prototype for #2134 
Related to stratis-storage/project#209

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mulkieran I've rebased and resolved conflicts to take advantage of the `extras` feature and new Makefile target. Let me know if we need to split it out into two different features when I request review. Doing a last little bit of review first.

**Comment by jbaublitz** [comment]
Rebased again.

**Comment by jbaublitz** [comment]
Rebased.

**Comment by jbaublitz** [comment]
Rebased and merge conflicts resolved.

**Comment by jbaublitz** [comment]
Blocked by #2163

**Comment by mulkieran** [comment]
@jbaublitz Please rebase when convenient.

**Comment by mulkieran** [comment]
I'll review this more thoroughly once #2163 clears.

File: src/bin/stratis-min/key.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::{fs::File, io, os::unix::io::AsRawFd};
+
+use libstratis::{
+    engine::{DeleteAction, KeyActions, MappingCreateAction, StratKeyActions},
+    stratis::{StratisError, StratisResult},
+};
+
+pub fn key_set(key_desc: &str, keyfile_path: Option<&str>, no_tty: bool) -> StratisResult<()> {
```
Comment (mulkieran) [inline src/bin/stratis-min/key.rs:12]: Please add some docs for this method.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Two minor requests...

**Comment by jbaublitz** [comment]
@mulkieran I've addressed the two concerns you had.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks good! Please close this and open a final PR.

**Pull request closed without merging**